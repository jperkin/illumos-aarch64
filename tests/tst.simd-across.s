/*
 * This file and its contents are supplied under the terms of the
 * Common Development and Distribution License ("CDDL"), version 1.0.
 * You may only use this file in accordance with the terms of version
 * 1.0 of the CDDL.
 *
 * A full copy of the text of the CDDL should have accompanied this
 * source.  A copy of the CDDL is also available via the Internet at
 * http://www.illumos.org/license/CDDL.
 */

/*
 * Copyright 2017 Joyent, Inc.
 */

/*
 * C3.6.4 AdvSIMD across lanes
 */
.macro simd_across_hsd op
	\op	h0, v0.8b
	\op	h31, v31.16b
	\op	s0, v0.4h
	\op	s31, v31.8h
	\op	d31, v31.4s
.endm
.macro simd_across_bhs op
	\op	b0, v0.8b
	\op	b31, v31.16b
	\op	h0, v0.4h
	\op	h31, v31.8h
	\op	s31, v31.4s
.endm
.macro simd_across_s op
	\op	s31, v31.4s
.endm

.text
.globl libdis_test
.type libdis_test, @function
libdis_test:
	simd_across_hsd saddlv
	simd_across_bhs smaxv
	simd_across_bhs sminv
	simd_across_bhs addv
	simd_across_hsd uaddlv
	simd_across_bhs umaxv
	simd_across_bhs uminv
	simd_across_s fmaxnmv
	simd_across_s fmaxv
	simd_across_s fminnmv
	simd_across_s fminv
.size libdis_test, [.-libdis_test]
