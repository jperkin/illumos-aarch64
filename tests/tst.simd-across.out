    libdis_test:      00 38 30 0e  saddlv h0, v0.8b
    libdis_test+0x4:  ff 3b 30 4e  saddlv h31, v31.16b
    libdis_test+0x8:  00 38 70 0e  saddlv s0, v0.4h
    libdis_test+0xc:  ff 3b 70 4e  saddlv s31, v31.8h
    libdis_test+0x10: ff 3b b0 4e  saddlv d31, v31.4s
    libdis_test+0x14: 00 a8 30 0e  smaxv b0, v0.8b
    libdis_test+0x18: ff ab 30 4e  smaxv b31, v31.16b
    libdis_test+0x1c: 00 a8 70 0e  smaxv h0, v0.4h
    libdis_test+0x20: ff ab 70 4e  smaxv h31, v31.8h
    libdis_test+0x24: ff ab b0 4e  smaxv s31, v31.4s
    libdis_test+0x28: 00 a8 31 0e  sminv b0, v0.8b
    libdis_test+0x2c: ff ab 31 4e  sminv b31, v31.16b
    libdis_test+0x30: 00 a8 71 0e  sminv h0, v0.4h
    libdis_test+0x34: ff ab 71 4e  sminv h31, v31.8h
    libdis_test+0x38: ff ab b1 4e  sminv s31, v31.4s
    libdis_test+0x3c: 00 b8 31 0e  addv b0, v0.8b
    libdis_test+0x40: ff bb 31 4e  addv b31, v31.16b
    libdis_test+0x44: 00 b8 71 0e  addv h0, v0.4h
    libdis_test+0x48: ff bb 71 4e  addv h31, v31.8h
    libdis_test+0x4c: ff bb b1 4e  addv s31, v31.4s
    libdis_test+0x50: 00 38 30 2e  uaddlv h0, v0.8b
    libdis_test+0x54: ff 3b 30 6e  uaddlv h31, v31.16b
    libdis_test+0x58: 00 38 70 2e  uaddlv s0, v0.4h
    libdis_test+0x5c: ff 3b 70 6e  uaddlv s31, v31.8h
    libdis_test+0x60: ff 3b b0 6e  uaddlv d31, v31.4s
    libdis_test+0x64: 00 a8 30 2e  umaxv b0, v0.8b
    libdis_test+0x68: ff ab 30 6e  umaxv b31, v31.16b
    libdis_test+0x6c: 00 a8 70 2e  umaxv h0, v0.4h
    libdis_test+0x70: ff ab 70 6e  umaxv h31, v31.8h
    libdis_test+0x74: ff ab b0 6e  umaxv s31, v31.4s
    libdis_test+0x78: 00 a8 31 2e  uminv b0, v0.8b
    libdis_test+0x7c: ff ab 31 6e  uminv b31, v31.16b
    libdis_test+0x80: 00 a8 71 2e  uminv h0, v0.4h
    libdis_test+0x84: ff ab 71 6e  uminv h31, v31.8h
    libdis_test+0x88: ff ab b1 6e  uminv s31, v31.4s
    libdis_test+0x8c: ff cb 30 6e  fmaxnmv s31, v31.4s
    libdis_test+0x90: ff fb 30 6e  fmaxv s31, v31.4s
    libdis_test+0x94: ff cb b0 6e  fminnmv s31, v31.4s
    libdis_test+0x98: ff fb b0 6e  fminv s31, v31.4s
